euphonic/plot.py

```diff
@@ -34,193 +42,288 @@ def plot_dispersion(phonons, btol=10.0, *args, **kwargs):
+    to avoid creating a vertical line
+    """
+
+    if isinstance(spectra, Spectrum1D):
+        return _plot_1d_core(Spectrum1DCollection.from_spectra([spectra]),
+                             ax=ax, **mplargs)
+
+    try:
+        assert isinstance(spectra, Spectrum1DCollection)
+    except AssertionError:
+        raise TypeError("spectra should be a Spectrum1D or "
+                        "Spectrum1DCollection")
+
+    # Find where there are two identical x_data points in a row
+    breakpoints = (np.where(spectra.x_data.magnitude[:-1]
+                            == spectra.x_data.magnitude[1:])[0]
+                   + 1).tolist()
+    breakpoints = [0] + breakpoints + [None]
+
+    for spectrum in spectra:
+        # Plot each line in segments, keeping colour consistent
+        for x0, x1 in zip(breakpoints[:-1], breakpoints[1:]):
+            if x0 == 0:
+                color = None
+            else:
+                color = p[-1].get_color()
+
+            p = ax.plot(spectrum._get_bin_centres('x').magnitude[x0:x1],
+                        spectrum.y_data.magnitude[x0:x1],
+                        color=color, **mplargs)
+
+    ax.set_xlim(left=min(spectra.x_data.magnitude),
+                right=max(spectra.x_data.magnitude))
+
+    _set_x_tick_labels(ax, spectra.x_tick_labels, spectra.x_data)
+
+
+def plot_1d(spectra: Union[Spectrum1D,
+                           Spectrum1DCollection,
+                           Sequence[Spectrum1D],
+                           Sequence[Spectrum1DCollection]],
+            title: str = '',
+            x_label: str = '',
+            y_label: str = '',
+            y_min: float = None,
+            y_max: float = None,
+            labels: Optional[List[str]] = None,
+            **line_kwargs) -> Figure:
     """
     Creates a Matplotlib figure for a Spectrum1D object, or multiple
     Spectrum1D objects to be plotted on the same axes
 
     Parameters
     ----------
-    spectra : Spectrum1D or list of Spectrum1D
-        Containing the 1D data to plot. Note only the x_tick_labels in
-        the first spectrum in the list will be used
-    title : string, optional
+    spectra
+        1D data to plot. Spectrum1D objects contain a single line, while
+        Spectrum1DCollection is suitable for plotting multiple lines
+        simultaneously (e.g. band structures).
+
+        Data split across several regions should be provided as a sequence of
+        spectrum objects::
+
+            [Spectrum1D, Spectrum1D, ...]
+
+        or::
+
+            [Spectrum1DCollection, Spectrum1DCollection, ...]
+
+        Where each segment will be plotted on a separate subplot. (This
+        approach is mainly used to handle discontinuities in Brillouin-zone
+        band structures, so the subplot widths will be based on the x-axis
+        ranges.)
+
+        A singular Spectrum1D or Spectrum1DCollection will be automatically
+        split into segments if btol was set.
```

Comment:
I don't think this comment is true, btol doesn't seem to be passed through anywhere?

Reply:
Indeed, btol isn't even accepted as an argument here.

We had discussed providing this as a convenience. But I'd be happy to leave it out; `plot_1d(my_spectrum.split())` is not such an unreasonable syntax, and it keeps the code cleaner.

Reply:
Sure, I'd be happy with that. Just need to make sure we mention the loss of the `btol` argument (and the `ratio` argument for `plot_2d`) in the changelog